source/includes/_security.md

```diff
@@ -1,5 +1,132 @@
+
+[https://jwt.io/](https://jwt.io/) is a great resource not only for learning about JSON web tokens, but also for parsing a JWT value into its distinct parts to see how it is constructed. Try taking the example JWT here and pasting it into the form at [https://jwt.io/](https://jwt.io/) to see how the token is constructed.
+
+<aside class="notice">
+TODO: Need to propose how the EHR public key is discovered. Preference would be a further extension in the EHR FHIR server Conformance resource.
+</aside>
+
+<aside class="notice">
+TODO: Need to outline the risks of noalg signatures
+</aside>
+
+<aside class="notice">
+TODO: Need to expose client_id and scopes in Discovery
+</aside>
+
+### Mutual TLS
+
+[Mutual TLS](https://en.wikipedia.org/wiki/Mutual_authentication) (mTLS) can be used between an EHR and CDS Service and that would allow the CDS Service to establish trust of the EHR. However, if mTLS is used, this should be in addition to using JSON web tokens to establish trust of the EHR. As mTLS is not well supported across all platforms and technologies, it is not the standard means of establishing trust with the EHR.
+
+## FHIR Resource Access
+
+The CDS Service is able to use the FHIR server of the EHR to obtain any additional data it needs in order to perform its decision support. This is similar to SMART on FHIR where the SMART app can obtain additional data via the provided FHIR server.
+
+Like SMART on FHIR, CDS Hooks requires that access to the FHIR server be controlled by an Authorization server utilizing the OAuth 2 framework. Thus, the CDS Service is able to consume the given FHIR server via an access (bearer) token just like a SMART app. While CDS Hooks shares the underlying technical framework and standards as SMART on FHIR, there are very important differences between SMART and CDS Hooks.
+
+### Obtaining an Access Token
+
+In SMART on FHIR, the SMART app requests and ultimately obtains an access token from the Authorization server using the SMART launch workflow. This process utilizes the authorization code grant model as defined by the OAuth 2.0 Authorization Framework in [rfc6749](https://tools.ietf.org/html/rfc6749).
+
+With CDS Hooks, the EHR provides the access token directly in the request to the CDS Service. Thus, the CDS Service does not need to request the token from the authorization server as a SMART app would. This is done purely for performance reasons as the authorization code grant model in OAuth 2 involves several HTTPS calls and redirects. In contrast to a SMART app, a CDS Service may be invoked many times during a workflow. Going through the authorization code grant model on every hook invocation would likely result in a slow performing CDS Service due to the authorization overhead.
+
+```json
+{
+  "fhirAuthorization" : {
+    "access_token" : "some-opaque-fhir-access-token",
+    "token_type" : "code",
```

Comment:
Can these fields be changed to be consistent with the "fhirAuthorization", "hookInstance", and other fields that do not use an underscore?

Reply:
Hey @robs16 , these properties names are intentionally the same as those defined by SMART and OAuth -- which do use underscores. 

What problem are the underscores causing for you?

Isaac

http://www.hl7.org/fhir/smart-app-launch/

Reply:
it doesn't cause an issue - its just inconsistent. It sounds like there is a good reason for the inconsistency though, so I'll retract my objection.